db/db_impl/db_impl.cc

```diff
@@ -453,7 +453,7 @@ Status DBImpl::ResumeImpl(DBRecoverContext context) {
   if (shutdown_initiated_) {
     s = Status::ShutdownInProgress();
   }
-  if (s.ok()) {
+  if (s.ok() && context.flush_after_recovery) {
```

Comment:
Can this scenario be avoided if we check ```shutdown_initiated_``` here?

Reply:
I think it would be the same as the existing check in line 453 (in the same critical section). I don't think it would work since db mutex is released/re-acquired several times in `FlushAllColumnFamilies()`.

Reply:
Ok, that's true. Its dropping the locking before scheduling the flush.